src/main/kotlin/org/janelia/saalfeldlab/labels/blocks/n5/LabelBlockLookupFromN5.kt

```diff
@@ -55,103 +55,125 @@ class LabelBlockLookupFromN5(
 			return map
 		}
 
-		private fun toBytes(map: Map<Long, Array<Interval>>): ByteArray {
-			val sizeInBytes = map.values.stream().mapToInt { java.lang.Long.BYTES + Integer.BYTES + SINGLE_ENTRY_BYTE_SIZE * it.size }.sum()
+		private fun toBytes(map: TLongObjectHashMap<Array<Interval>>): ByteArray {
+			val sizeInBytes = map.valueCollection().stream().mapToInt { java.lang.Long.BYTES + Integer.BYTES + SINGLE_ENTRY_BYTE_SIZE * it.size }.sum()
 			val bytes = ByteArray(sizeInBytes)
 			val bb = ByteBuffer.wrap(bytes)
-			for (entry in map) {
-				bb.putLong(entry.key)
-				bb.putInt(entry.value.size)
-				for (interval in entry.value) {
+			map.forEachEntry { key, value ->
+				bb.putLong(key)
+				bb.putInt(value.size)
+				for (interval in value) {
 					bb.putLong(interval.min(0))
 					bb.putLong(interval.min(1))
 					bb.putLong(interval.min(2))
 					bb.putLong(interval.max(0))
 					bb.putLong(interval.max(1))
 					bb.putLong(interval.max(2))
 				}
+				true
 			}
 			return bytes
 		}
 	}
 
 	private val attributes = mutableMapOf<Int, DatasetAttributes>()
 
+	private val cache = TIntObjectHashMap<TLongObjectHashMap<TLongObjectHashMap<Array<Interval>>>>() // scale -> block index -> id -> intervals
```

Comment:
Why not use an ImgLib2 cache with a custom key like
```kotlin
data class Key(
    val scale: Int,
    val blockIndex: Long,
    val id: Long)
```
?
That could also be used for invalidation, then.

Reply:
Switched to cache key `(scale, blockId)` and some extra conversion between cache keys and input lookup keys `(scale, labelId)`